r/vignettes/fs.Rmd

```diff
@@ -128,3 +128,74 @@ s3://minioadmin:minioadmin@?scheme=http&endpoint_override=localhost%3A9000
 
 Among other applications, this can be useful for testing out code locally before
 running on a remote S3 bucket.
+
+## Non-AWS S3 cloud alternatives (DigitalOcean, IBM, Alibaba, and others)
+
+*This section adapts some elements from [Analyzing Room Temperature Data](https://www.jaredlander.com/2021/03/analyzing-room-temperature-data/#getting-the-data) by Jared Lander.*
+
+If you are using any Amazon S3 Compliant Storage Provider, such as AWS, Alibaba, 
+Ceph, DigitalOcean, Dreamhost, IBM COS, Minio, or others, you can connect to it 
+with `arrow` by using the `S3FileSystem` function as for the case of using 
+MinIO locally. Please note that the use of DigitalOcean here is just an example, as 
+it can be any other S3 compatible service.
+
+At the begininning of this vignette we used:
+
+```r
+june2019 <- SubTreeFileSystem$create("s3://ursa-labs-taxi-data/2019/06")
+```
+
+Which connects to AWS, and the same can be adapted for other providers, For 
+instructional purposes, we provide [nyc-taxi.sfo3.digitaloceanspaces.com](https://nyc-taxi.sfo3.digitaloceanspaces.com), 
+which is a public storage with the NYC taxi data used in
+[Working with Arrow Datasets and dplyr](dataset.html).
+
+To connect to this space, you only need to adapt the code from the previous
+section:
+
+```r
+space <- arrow::S3FileSystem$create(
+  anonymous = TRUE,
+  scheme = "https",
+  endpoint_override = "sfo3.digitaloceanspaces.com"
+)
+```
+
+The space that we are using space allows anonymous access, but if you were to 
+connect to a private space (i.e. with sensitive data), you would need to 
+provide a token, say:
+
+```r
+space <- arrow::S3FileSystem$create(
+  access_key = Sys.getenv('DO_ARROW_TAXI_TOKEN'),
+  secret_key = Sys.getenv('DO_ARROW_TAXI_SECRET'),
+  scheme = "https",
+  endpoint_override = "sfo3.digitaloceanspaces.com"
+)
+```
```

Comment:
Instead of using digital ocean as our "here is an example of an alternative" let's lean into the minio example that is already there. That has a few nice advantages: 
* anyone can install minio without needing to put data in a service
* we can test against minio to confirm that this works
* we don't need to maintain a new data source

Reply:
this is great

Reply:
how about a ... "just as an example..."?

Reply:
i'll send a coutner proposal for this